### ISSUE TITLE
Can't Create/Manage Firewall Rules at Node/Cluster Level

### ISSUE THREAD

**Issue by simoncaron** [bpg/terraform-provider-proxmox#1176]
**Describe the bug**
Can't configure interface for firewall rules at node/cluster level.

**To Reproduce**

1. Create a new firewall rule at node level ex:
```
resource "proxmox_virtual_environment_firewall_rules" "pve01_datacenter_firewall_rules" {
  node_name = "pve01"

  # PVE Admin Interface
  rule {
    type   = "in"
    action = "ACCEPT"
    iface  = "vmbr0"
    dport = "8006"
    proto = "tcp"
    comment = "PVE Admin Interface"
  }
}
```
2. Apply configuration

**Expected behavior**
A new rule is created at node level
![Capture d’écran du 2024-04-01 21-30-09](https://github.com/bpg/terraform-provider-proxmox/assets/8635747/c497fde0-f4a6-4ec8-8892-0ed584fa4eff)

**Current behavior**
Terraform apply fail with the following error:

> ╷
> │ Error: invalid value for iface (Must be a valid VM/Container iface key, e.g. 'net0')
> │ 
> │   with proxmox_virtual_environment_firewall_rules.pve01_node_firewall_rules,
> │   on firewall_pve01.tf line 41, in resource "proxmox_virtual_environment_firewall_rules" "pve01_node_firewall_rules":
> │   41:     iface  = "vmbr0"
> │ 
> ╵
> error: Recipe `terraform` failed with exit code 1

**Additional informations**
The regex which is validating the interface name for VMs/Container shouldn't be applied for Node/Cluster fw rules since interface names are not standard at these levels.

**Comment by simoncaron** [comment]
Cluster/Datacenter level would do the trick for me but even after removing the `node_name` attribute, I'm still getting the same iface validation error... :)

Looks like the validation with the regex `net\d+` is called [systematically here](https://github.com/bpg/terraform-provider-proxmox/blob/13b25df00548fc0493cd56be58d7e1913761cee9/proxmoxtf/resource/firewall/rules.go#L120), even on cluster rules.

**Comment by midnight-pm** [comment]
Hi @bpg,

I can confirm that I am experiencing the same error as @simoncaron.

When attempting to create Firewall Rules at the cluster level, I receive the same error that they indicated in [their initial report](https://github.com/bpg/terraform-provider-proxmox/issues/1176#issue-2219397042).

As an example, assume a block that appears as follows:

```
resource "proxmox_virtual_environment_firewall_rules" "inbound" {

  provider = "proxmox-bpg"

  # Interface: vmbr0 (Public Interface)
  rule {
    security_group = "foo"
    comment        = "Allow Rule for Public Interface for Accepting Connections from foo"
    iface          = "vmbr0"
    enabled        = true
  }

  # Interface: vmbr1 (Private Interface)
  rule {
    security_group = "bar"
    comment        = "Allow Rule for Public Interface for Accepting Connections from bar"
    iface          = "vmbr1"
    enabled        = true
  }

  [...]
}
```

When `terraform plan` is executed, the following occurs:

```
╷
│ Error: invalid value for iface (Must be a valid VM/Container iface key, e.g. 'net0')
│ 
│   with proxmox_virtual_environment_firewall_rules.inbound,
│   on proxmox-firewall.tf line 66, in resource "proxmox_virtual_environment_firewall_rules" "inbound":
│   66:     iface          = "vmbr0"
│ 
╵
╷
│ Error: invalid value for iface (Must be a valid VM/Container iface key, e.g. 'net0')
│ 
│   with proxmox_virtual_environment_firewall_rules.inbound,
│   on proxmox-firewall.tf line 75, in resource "proxmox_virtual_environment_firewall_rules" "inbound":
│   75:     iface          = "vmbr1"
│ 
╵
```

Notably, this is in spite of `node_name` not being present as an argument in the configuration block above. For that matter, `vm_id` and `container_id` are also not specified.

The same behavior is presented by the provider if `node_name` _is_ present regardless of whether or not it has a value of `null` (e.g.: `node_name = null`) or is given the value of an empty string, e.g.: `node_name = ""`.

In short, it appears that the provider attempts to treat the configuration as if it should be assigned to a VM or to a Container regardless of whether or not the parameter `node_name` is present in the configuration, and accordingly issues an error.
